### ISSUE TITLE
Dimmer only closes first time with modal

### ISSUE THREAD

**Issue by ThatsIT** [Semantic-Org/Semantic-UI#1970]
I have a modal, that opens ok, closes ok, then when reopened and closed again, the dimmer stays. This is using "hide" from a button click, if I click the dimmer itself it seems to work. 
When opening the second time there is a bit of flashing. 

my code, html incomplete as it is inserted via ajax, see lower down 

```html
<div id="empty-form-modal" class="ui modal">
    <i class="close icon"></i>
    <div class="header">

    </div>
    <div id="content-box">

    </div>
</div>
```

```javascript
var EmptyFormModel = (function () {
    function EmptyFormModelx(url, data, title, width, height) {
        this.url = url;
        this.data = data;
        this.title = title;
        this.width = width;
        this.height = height;
        this.eModal = $("#empty-form-modal");
        $("#empty-form-modal .header").html(this.title);
    }
    EmptyFormModelx.prototype.open = function () {
        var p = $.get(this.url, this.data, function (data) {
            $("#empty-form-modal #content-box").html(data);
        }, "html");
        p.done(function () {
            $("#empty-form-modal #save-button").off("click.emptyModalSave");
            $("#empty-form-modal #save-button").on("click.emptyModalSave", function () {
                var f = $("#empty-form-modal").find("form");
                $.validator.unobtrusive.parse(f);
                if (f.valid()) {
                    $.post(f.attr("action"), f.serialize(), function (data) {
                        $("#empty-form-modal #content-box").html(data);
                    }, "html");
                }
            });
            $("#empty-form-modal #close-button").off("click.emptyModalClose");
            $("#empty-form-modal #close-button").on("click.emptyModalClose", function () {
                $("#empty-form-modal").modal("hide");
                this.eModal = null;
            });
            $("#empty-form-modal").modal("show");
        });
        return p;
    };
    return EmptyFormModelx;
})();
```

html returned by ajax

```html
    <div id="size-modal" class="ui two column grid content">
        <div class="one column row">
            <div class="column">
                <h3>@Model.Product.Name</h3>
            </div>
        </div>
        <div class="column">
            <img class="img-responsive" alt="@Model.Product.Name" src="/products/productimage/@Model.Product.ImageId" />
        </div>
        <div class="column">
            <h4>Select Size</h4>
            @For Each size As EF.Size In Model.Sizes
                @<label>@size.Name  @Html.RadioButton("Size", size.SizeId, False)</label>
                @<br />
            Next
            <div class="alert alert-warning display-none" id="size-alert">Please select size</div>
        </div>

    </div>

    <div class="actions">
         <div id="close-button" class="ui black button">Close</div>
        <div id="save-sizes-button" class="ui positive right labeled icon button">Add to cart @ThatsFontAwesome.GetFont

(Function(f) f.CartPlus).Size(Function(f) f.Large2x).Render</div>
    </div>
```

Thanks

**Comment by jlukic** [comment]
Please create a jsfiddle that reproduces your issue and a list of reproducible steps.

**Comment by ThatsIT** [comment]
Jack the jsfiddle worked, so I started a process of elimination till I got down to just the bare semantic-ui  code and still had problem. I then replaced the refs to semantic css and js to those on the semantic-ui.com site and hay presto it worked.

The files I had I got out of the from the site download zip dist folder (min) and added them to my mvc.net application

I have now used bower on Visual Studio 2015 to get the files and they work.
Thanks
